### PR TITLE
[Wait for #1815][lstmcell] Enable multi inout for lstmcell, zoneout lstmcell

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1:     [zoneout lstmcell] support multi in/out
    
     - Refactoring zoneout lstmcell layer to support multi in/out (3 input / 2output)
     - Regenerate zoneout testcase for multi in/out
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 2:     [bugfix] zoneout lstmcell regenerate mask
    
     - Current implementation regenetate the zoneout mask in calcGradient.
        Fix it to reuse the zoneout mask and remove renerating the mask.
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 3:     [lstmcell] support multi in/out
    
     - Refactoring lstmcell layer to support multi in/out (3 input / 2 output)
     - Regenerate lstmcell testcase for multi in/out
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: test/input_gen/genModelsRecurrent_v2.py [hunk 1]
```diff
@@ -197,147 +199,165 @@ def forward(self, inputs, labels):
         name="lstm_stacked",
     )
 
+    unroll_for, num_lstm, state_num, batch_size, unit, feature_size, iteration, hidden_state_zoneout_rate, cell_state_zoneout_rate = [2, 1, 2, 1, 2, 2, 2, 0.0, 0.0]
     record_v2(
-        ZoneoutLSTMStacked(batch_size=1, unroll_for=2, num_lstm=1, hidden_state_zoneout_rate=0.0, cell_state_zoneout_rate=0.0),
-        iteration=2,
-        input_dims=[(1, 2)],
-        label_dims=[(1, 2, 2)],
+        ZoneoutLSTMStacked(batch_size=batch_size, unroll_for=unroll_for, num_lstm=num_lstm, hidden_state_zoneout_rate=hidden_state_zoneout_rate, cell_state_zoneout_rate=cell_state_zoneout_rate),
```
Comment (zhoonit) [inline test/input_gen/genModelsRecurrent_v2.py:252]: recommend to use functools partial if this setup is being repeatedly used. For now it looks fine :)

File: nntrainer/compiler/recurrent_realizer.cpp [hunk 2]
```diff
@@ -158,7 +157,6 @@ static void propagateTimestep(LayerNode *node, unsigned int time_step,
   auto is_recurrent_type = [](LayerNode *node) {
     return node->getType() == RNNCellLayer::type ||
            node->getType() == LSTMLayer::type ||
-           node->getType() == LSTMCellLayer::type ||
            node->getType() == ZoneoutLSTMCellLayer::type ||
```
Comment (zhoonit) [inline nntrainer/compiler/recurrent_realizer.cpp:160]: Zoneout should be removed as well?
If this is working, does zoneout still takes time_steps?

Comment (zhoonit) [reply nntrainer/compiler/recurrent_realizer.cpp:160]: Discussed offline that this is required for now as zoneout mask is being shared.

Later, we can remove timestep as zoneout mask will be held individually.

**Review by zhoonit — APPROVED** [review]
LGTM after clang fix.

Nice workline :)

File: nntrainer/compiler/recurrent_realizer.cpp (hunk 2 above)
Comment (lhs8928) [reply nntrainer/compiler/recurrent_realizer.cpp:160]: As discussed offline, for now zoneout needs timestep for the layer unittest

**Review by kparichay — APPROVED** [review]
overall LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**